sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/JoinQueryDetector.scala

```diff
@@ -164,9 +164,9 @@ class JoinQueryDetector(sparkSession: SparkSession) extends Strategy {
     val relationship = s"ST_$spatialPredicate"
 
     matchExpressionsToPlans(a, b, left, right) match {
-      case Some((planA, planB, _)) =>
+      case Some((planA, planB, swappedLeftAndRight)) =>
         logInfo(s"Planning spatial join for $relationship relationship")
-        RangeJoinExec(planLater(planA), planLater(planB), a, b, spatialPredicate, extraCondition) :: Nil
+        RangeJoinExec(planLater(planA), planLater(planB), a, b, swappedLeftAndRight, spatialPredicate, extraCondition) :: Nil
```

Comment:
Can these just be handled like the broadcast join below where the swapping is handled here with pattern matching instead of pushed into the Exec node?

Reply:
Exec node has to handle the swapping somewhere, unless we invert the spatial predicate in the join query detector.

Broadcast join exec node [resolves if left and right were swapped](https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/BroadcastIndexJoinExec.scala#L117-L122) using `indexBuildSide` and `windowJoinSide`, it does not seems to be more intuitive than simply pushing the swapping indicator to the `TraitJoinQueryExec` node.

Inverting the spatial predicate seems to be a viable approach, I'll submit changes to this PR if you're OK with it.

Reply:
@Kimahriman Adam, does @Kontinuation 's approach sound good to you? :-)

Reply:
I'm trying to wrap my head around this a little bit. By inverting the spatial predicate do you mean like rewritting it? As in `ST_Contains(a, b)` -> `ST_Within(b, a)` or something like that?

What would happen if you just swapped the left and right shapes if needed? Like
```scala
    matchExpressionsToPlans(a, b, left, right) match {
      case Some((_, _, swapped)) =>
        logInfo(s"Planning spatial join for $relationship relationship")
        if (swapped) {
          RangeJoinExec(planLater(left), planLater(right), b, a, spatialPredicate, extraCondition) :: Nil
        } else {
          RangeJoinExec(planLater(left), planLater(right), a, b, spatialPredicate, extraCondition) :: Nil
        }
      case None =>
        logInfo(
          s"Spatial join for $relationship with arguments not aligned " +
            "with join relations is not supported")
        Nil
    }
```

I'm trying to think of why that wouldn't work